libafl/src/events/llmp/restarting.rs

```diff
@@ -300,67 +311,95 @@ where
     }
 }
 
-impl<E, EMH, I, S, SHM, SP, Z> EventProcessor<E, S, Z>
-    for LlmpRestartingEventManager<EMH, I, S, SHM, SP>
+impl<EMH, I, S, SHM, SP> EventProcessor<I, S> for LlmpRestartingEventManager<EMH, I, S, SHM, SP>
 where
-    E: HasObservers,
-    E::Observers: DeserializeOwned,
     EMH: EventManagerHooksTuple<I, S>,
     I: DeserializeOwned + Input,
     S: HasImported + HasCurrentTestcase<I> + HasSolutions<I> + Stoppable + Serialize,
     SHM: ShMem,
     SP: ShMemProvider<ShMem = SHM>,
-    Z: ExecutionProcessor<Self, I, E::Observers, S> + EvaluatorObservers<E, Self, I, S>,
 {
-    fn process(&mut self, fuzzer: &mut Z, state: &mut S, executor: &mut E) -> Result<usize, Error> {
-        let res = {
-            // TODO: Get around local event copy by moving handle_in_client
-            let self_id = self.llmp.sender().id();
-            let mut count = 0;
-            while let Some((client_id, tag, flags, msg)) = self.llmp.recv_buf_with_flags()? {
-                assert_ne!(
-                    tag, _LLMP_TAG_EVENT_TO_BROKER,
-                    "EVENT_TO_BROKER parcel should not have arrived in the client!"
-                );
-
-                if client_id == self_id {
-                    continue;
-                }
+    fn receive(&mut self, state: &mut S) -> Result<Vec<(Event<I>, bool)>, Error> {
+        let mut event_vec = vec![];
```

Comment:
Let's avoid allocation for every incoming message. Return them one by one instead, like Option<Event<I>>